### DISCUSSION TITLE
Is the 'enable speedhack detect' option meant to completely disable 'Rapid Hack Detected' kicks?

### DISCUSSION THREAD

**Discussion by lokka30** [piqueserver/piqueserver#617] [Q&A] [answered]
Is the 'enable speedhack detect' option meant to completely disable 'Rapid Hack Detected' kicks?

I have disabled this option on my server TheHallway, yet unfortunately people are still getting kicked for this.

I didn't create a bug report since I am unsure if this is intended or not.

Thank you!

**Comment by NotAFile** [comment, marked as answer]
Hey Lokka,

Speed hack refers to fast movement, while rapid hack refers to using tools too fast. I think they're separate because they used to be extension scripts. Unfortunately there's no config setting to turn off rapid hack detection right now.

That could be added as a new setting, or controlled by `detect_speedhack`. Is there any case where you'd want one but not the other?

Ideally, we would just improve the detection to cause less false positives instead though. If anyone wants to help collect data to tune the thresholds it would be very appreciated. Sharing the output of `grep "RAPID HACK:" path/to/logfile` might already be enough there.

**Reply by lokka30** [reply]
Hi NotAFile, cheers for the answer -

I believe that it is best that it is added as a new setting: `detect_rapid_tool_hack`, or if simpler, `detect_toolhack`. In this case, if speedhack is causing false positives (for example) but the toolhack detection isn't, then one can be left enabled. :)

I should be able to send over our rapid hack logs. Where abouts would you prefer me add them? In a separate discussion post like this one, however posting a new reply with a new pastebin link for each log paste? Or, a new channel on the Discord? - some recommendations.

I have fiddled with the Pyspades code to try disable rapid hack detection, but it was fruitless. The likely case is since I have little idea what I am actually doing with Python, I've probably touched the wrong code. 😁 

Have a good one. 👍🏻
